app/controllers/items_controller.rb

```diff
@@ -49,7 +49,11 @@ def create
       # with the error + the invalid parameters
       @item = current_organization.items.new(item_params)
 
-      flash[:error] = "An item with that name already exists (could be an inactive item)."
+      if result.error.is_a?(ActiveRecord::RecordInvalid) && create.send(:item_params)[:name].present?
```

Comment:
With this conditional both error messages are kept in the case of other scenarios. Are there any other ways to solve this?

Reply:
Something like `current_organization.items.exists?(name: item_params[:name])` would work. But it is not case-insensitive

Reply:
Additionally, I meant to change `result.error.is_a?(ActiveRecord::RecordInvalid) && create.send(:item_params)[:name].present?`
to `result.error.is_a?(ActiveRecord::RecordInvalid) && item_params[:name].present?` before my push.

Reply:
As long as we're trying to send a different error message for this specific situation don't see any other way of doing this.

Reply:
From what @vsrnth mentioned one shows the new error message when the uniqueness check fails and the old one when that's not the case. Like when the creation of a new item with empty attributes occurs (item_system_spec:25) So essentially when the form is blank and a user tries to save and create a new item.

Reply:
I get that that's what it's doing - I just don't understand why.

Reply:
hmm instead of something like `flash[:error] = result.error.message.full_messages` and just using activerecord validation errors?